src/bokeh/core/serialization.py

```diff
@@ -441,12 +437,11 @@ def _encode_other(self, obj: Any) -> AnyRep:
         if np.issubdtype(type(obj), np.bool_):
             return self._encode_bool(bool(obj))
 
-        if _HAS_PANDAS:
-            module = type(obj).__module__
-            if module is not None and module.startswith("pandas"):
-                pd = import_optional("pandas")
-                if pd and isinstance(obj, (pd.Series, pd.Index)):
-                    return self._encode_ndarray(transform_series(obj))
+        import pandas as pd
+        module = type(obj).__module__
+        if module is not None and module.startswith("pandas"):
```

Comment:
This was guarding the optional import. Either remove it or move the import under this condition. Perhaps the later, unless we want to import pandas on serialization when no pandas objects are involved.

Reply:
I suppose unlike other places where deferred imports are a result of direct usage involving pandas, serialization will be used regardless. I'll put back the guard but with a comment explaining why it is there.